planning/behavior_path_planner_common/src/utils/drivable_area_expansion/static_drivable_area.cpp

```diff
@@ -227,7 +227,8 @@ double calcDistanceFromPointToSegment(
 
   // target_point is outside the segment.
   return std::min(
-    tier4_autoware_utils::calcDistance2d(a, p), tier4_autoware_utils::calcDistance2d(b, p));
+    tier4_autoware_utils::calcSquaredDistance2d(a, p),
+    tier4_autoware_utils::calcSquaredDistance2d(b, p));
```

Comment:
Is it okay to return a squared distance here ?
How about putting the `std::min` inside a call to `std::sqrt` ?

Reply:
In calcDistanceFromPointToSegment, if the condition if (0 <= dot_val && dot_val <= squared_segment_length) is met, I believe the non squared distance is returned? 

In that case, you would be comparing squared and non-squared distances on line 242 no? 

Maybe it is better to return squared distances on both cases an rename the function to calcSquaredDistanceFromPointToSegment?

Reply:
Sure :+1:
As Daniel said, you also need to make sure the `numerator / denominator` is a squared distance for consistency.

Reply:
@maxime-clem @danielsanchezaran Oh. I missed it. Thanks :+1:

Reply:
@maxime-clem I renamed the function and fixed process consistency in https://github.com/autowarefoundation/autoware.universe/pull/6054/commits/34f0c19e6d0341a9baa5744072e9b23bc15a3ca0. Could you check that...?

Reply:
Looks good thank you.